Documenting the repo move (#629)
actualbudget/docs: docs/actual-server-repo-move.md

# Actual Server repository move

> In February 2025 we will begin the process of merging the actual-server repository into the Actual workspace.

Reviewer: nit: if we plan to merge this PR only after the server merge, maybe reword to:
```suggestion
In February 2025 we merged the `actual-server` repository into the Actual workspace.
```

Reviewer: Ah I see, this is a temporary measure to prevent confusion. ff to ignore this suggestion for now then

The reasons for this change are as follows:

- Streamlined Development: Developers will only need to clone/sync one repo instead of two.
- Improved Debugging: It makes end-to-end debugging for server/client easier as they will be in the same workspace.
- Simplified Desktop App Packaging: Enables the desktop app to embed the sync server.

> - Ensuring code consistency and reducing the maintenance burden.

Reviewer: Also would add it makes the releasing process easier for us (as I understand we no longer need to publish the npm web package and hence deal with all the specific ordering of the release process related to that)
